substrate/client/network/benches/notifications_protocol.rs

```diff
@@ -25,55 +25,42 @@ use sc_network::{
 		FullNetworkConfiguration, MultiaddrWithPeerId, NetworkConfiguration, NonReservedPeerMode,
 		NotificationHandshake, Params, ProtocolId, Role, SetConfig,
 	},
-	service::traits::NotificationEvent,
+	service::traits::{NetworkService, NotificationEvent},
 	Litep2pNetworkBackend, NetworkBackend, NetworkWorker, NotificationMetrics, NotificationService,
-	Roles,
+	PeerId, Roles,
 };
 use sc_network_common::{sync::message::BlockAnnouncesHandshake, ExHashT};
-use sc_network_types::build_multiaddr;
 use sp_core::H256;
 use sp_runtime::traits::{Block as BlockT, Zero};
-use std::{
-	net::{IpAddr, Ipv4Addr, TcpListener},
-	str::FromStr,
-};
+use std::{sync::Arc, time::Duration};
 use substrate_test_runtime_client::runtime;
+use tokio::{sync::Mutex, task::JoinHandle};
 
-const MAX_SIZE: u64 = 2u64.pow(30);
-const SAMPLE_SIZE: usize = 50;
-const NOTIFICATIONS: usize = 50;
-const EXPONENTS: &[(u32, &'static str)] = &[
-	(6, "64B"),
-	(9, "512B"),
-	(12, "4KB"),
-	(15, "64KB"),
-	(18, "256KB"),
-	(21, "2MB"),
-	(24, "16MB"),
-	(27, "128MB"),
+const SMALL_PAYLOAD: &[(u32, usize, &'static str)] = &[
+	// (Exponent of size, number of notifications, label)
+	(6, 100, "64B"),
```

Comment:
Can we increase the number of notifications, especially for smaller payload sizes? Even on localhost, the TCP window control can influence the results quite a lot and for low count of small notifications we can get inaccurate results.

Ideally, we would like to transmit tens of megabytes for every notification size. I.e., we can keep constant (size) * (number) = 100 MB. In case it takes too long for smaller payload sizes, we can limit it to 15-30 seconds. And we can also decrease the number of criterion iterations for every size. 10 should be fine to have an estimate of the mean deviation

We can do this number picking in a follow up PR.